src/wam2layers/preprocessing/era5.py

```diff
@@ -185,45 +193,63 @@ def get_dp_pressurelevels(q, u, v, ps, qs, us, vs):
         v interpolated to new pressure levels,
         Pressure at the division between the two layers
     """
-    assert u.dims[0] == "level"  # move to better location
+    level_ax: int = u.get_axis_num("level")  # type: ignore
+    spatial_axes = u.get_axis_num(("latitude", "longitude"))
 
     p = u["level"].broadcast_like(u) * 100  # Pa
     p.attrs["units"] = "Pa"
 
+    _p = p.to_numpy()
+    _u = u.to_numpy()
+    _v = v.to_numpy()
+    _q = q.to_numpy()
+
     # Insert top of atmosphere values
-    # Assume wind at top same as values at lowest pressure, humidity at top 0
-    u = insert_level(u, u.isel(level=0), 0)
-    v = insert_level(v, v.isel(level=0), 0)
-    q = insert_level(q, 0, 0)
-    p = insert_level(p, 0, 0)
-
-    # Insert surface level values (at a high dummy pressure value)
-    u = insert_level(u, us, 110000)
-    v = insert_level(v, vs, 110000)
-    q = insert_level(q, qs, 110000)
-    p = insert_level(p, ps, 110000)
+    # Assume wind at top same as values at lowest pressure, humidity at top = 0
+    p0 = _p.mean(axis=spatial_axes).argmin()  # should be a cleaner way to do this.
```

Comment:
Why take the spatial mean?

Reply:
This is the line I wanted to improve. I'm not sure what the best way of taking the slice `_p[:,0,0]` is programmatically (i.e. with the `level_ax` variable... :thinking:

Reply:
What about `_p.argmin(axis=level_ax)`?

Reply:
Also, I think you want the slice `_p[0, :, :]`, right?

Reply:
I just need to know the index along the `level_ax` where the lowest pressure is.

```py
_p.argmin(axis=level_ax).flatten()[0]
```
Seems to work well.

```py
_p.argmin(axis=level_ax)
```
gives a 2d array, which I don't want here.